aries_vcx/src/libindy/utils/anoncreds.rs

```diff
@@ -852,19 +855,25 @@ pub async fn revoke_credential_local(
     rev_reg_id: &str,
     cred_rev_id: &str,
 ) -> VcxResult<()> {
-    let mut new_delta = libindy_issuer_revoke_credential(wallet_handle, tails_file, rev_reg_id, cred_rev_id).await?;
-    if let Some(old_delta) = get_rev_reg_delta_cache(wallet_handle, rev_reg_id).await {
-        new_delta = libindy_issuer_merge_revocation_registry_deltas(old_delta.as_str(), new_delta.as_str()).await?;
+    let mut new_delta_json = libindy_issuer_revoke_credential(wallet_handle, tails_file, rev_reg_id, cred_rev_id).await?;
+    info!("revoke_credential_local >>> new_delta_json: {}", new_delta_json);
+    if let Some(old_delta_json) = get_rev_reg_delta(wallet_handle, rev_reg_id).await {
+        info!("revoke_credential_local >>> old_delta_json: {}", old_delta_json);
+        new_delta_json = libindy_issuer_merge_revocation_registry_deltas(old_delta_json.as_str(), new_delta_json.as_str()).await?;
+        info!("revoke_credential_local >>> merged_delta_json: {}", new_delta_json);
     }
-    set_rev_reg_delta_cache(wallet_handle, rev_reg_id, &new_delta).await
+    set_rev_reg_delta(wallet_handle, rev_reg_id, &new_delta_json).await
 }
 
-pub async fn publish_local_revocations(wallet_handle: WalletHandle, pool_handle: PoolHandle, submitter_did: &str, rev_reg_id: &str) -> VcxResult<String> {
-    if let Some(delta) = get_rev_reg_delta_cache(wallet_handle, rev_reg_id).await {
-        match clear_rev_reg_delta_cache(wallet_handle, rev_reg_id).await {
-            Ok(_) => publish_rev_reg_delta(wallet_handle, pool_handle, &submitter_did, rev_reg_id, &delta).await,
-            Err(err) => Err(err),
+pub async fn publish_local_revocations(wallet_handle: WalletHandle, pool_handle: PoolHandle, submitter_did: &str, rev_reg_id: &str) -> VcxResult<()> {
+    if let Some(delta) = get_rev_reg_delta(wallet_handle, rev_reg_id).await {
+        publish_rev_reg_delta(wallet_handle, pool_handle, &submitter_did, rev_reg_id, &delta).await?;
+        info!("publish_local_revocations >>> rev_reg_delta published for rev_reg_id {}", rev_reg_id);
+        match clear_rev_reg_delta(wallet_handle, rev_reg_id).await {
+            Ok(_) => info!("publish_local_revocations >>> rev_reg_delta cache cleared for rev_reg_id {}", rev_reg_id),
+            Err(err) => error!("publish_local_revocations >>> rev_reg_delta cache clear failed for rev_reg_id {}: {}", rev_reg_id, err)
```

Comment:
Perhaps we should return an error in this case - it gets the library into an inconsistent state, which is quite severe